### PR TITLE
ImageEntry Thumbnails (and various UI improvements)

### LINKED ISSUE #170: Add thumbnail to ImageEntry on ETL
1. Use rasterio/GDAL to save a small thumbnail of all imagery to the `ImageEntry` model during the ETL routine 

2. provide a way to pull that image and overlay it in the GeoJS front end

@manthey, should I add a new argument to the endpoints such as `&thumbnail=true` with a default value `false` to pull the thumbnail as a part of the query results when they return rasters/other imagery?

### PR DESCRIPTION
**Pull request by banesullivan**
Resolve #170 

Todo:
- [x] Add ETL routine to extract thumbnail - this currently maps the first map with a Viridis colormap
- [x] Improve the thumbnail routine to use RGB channels in the image if present, greyscale if only one band, and or map the first band to a Viridis colormap in other rare cases
- [x] Show thumbnail on GeoJS map view in the `RasterEntry` detail view

------

Currently, the detail view for `RasterEntry` shows a map with a thumbnail overlain the bounds of the raster and a polygon of the non-null data footprint:

Screen Shot 2020-10-08 at 11 39 59 AM

Screen Shot 2020-10-08 at 11 50 38 AM

From this, we can see a minor mismatch of the non-null data polygon and the thumbnail. I'm not sure if this is simply a display issue or if the convex hull polygon might be incorrect.

### CODE REVIEW

File: rgd/geodata/templates/geodata/rasterentry_detail.html [hunk 1]
```diff
@@ -16,6 +16,55 @@ <h3><span>Raster Metadata</span></h3>
...
+
+    let img = new Image();
+    img.src = extents['img_url'];
+
+    var quad = layer.createFeature('quad')
+      .data([
+        {
+          ll: {x: extents.extent.xmin, y: extents.extent.ymin },
+          ur: {x: extents.extent.xmax, y: extents.extent.ymax },
+          // image: img,
```
Comment (banesullivan) [inline rgd/geodata/templates/geodata/rasterentry_detail.html:32]: This throws the error. The image URL is correct as far as I can tell (Creating an HTML element for that Image object renders the image on the page fine). GeoJs throws the error:

```
DOMException: Failed to execute 'texImage2D' on 'WebGLRenderingContext': The image element contains cross-origin data, and may not be loaded.
    at t.texture.setup (https://cdnjs.cloudflare.com/ajax/libs/geojs/0.20.0/geo.min.js:72:110112)
    at t.texture.bind (https://cdnjs.cloudflare.com/ajax/libs/geojs/0.20.0/geo.min.js:72:110473)
    at Object.<anonymous> (https://cdnjs.cloudflare.com/ajax/libs/geojs/0.20.0/geo.min.js:83:2166058)
    at Function.each (https://cdnjs.cloudflare.com/ajax/libs/geojs/0.20.0/geo.min.js:28:2956)
    at t.mapper._renderImageQuads [as render] (https://cdnjs.cloudflare.com/ajax/libs/geojs/0.20.0/geo.min.js:83:2166008)
    at t.renderer.render (https://cdnjs.cloudflare.com/ajax/libs/geojs/0.20.0/geo.min.js:72:85361)
    at t.renderWindow.render (https://cdnjs.cloudflare.com/ajax/libs/geojs/0.20.0/geo.min.js:72:91701)
    at t.viewer.render (https://cdnjs.cloudflare.com/ajax/libs/geojs/0.20.0/geo.min.js:72:102404)
    at e._renderFrame (https://cdnjs.cloudflare.com/ajax/libs/geojs/0.20.0/geo.min.js:39:142852)
    at H (https://cdnjs.cloudflare.com/ajax/libs/geojs/0.20.0/geo.min.js:83:181219)
```

Comment (banesullivan) [reply rgd/geodata/templates/geodata/rasterentry_detail.html:32]: Fixed by:

```js
img.crossOrigin = 'anonymous'
```
https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/crossOrigin

**Comment by banesullivan** [comment]
We may also want to add a way for the `RasterEntry` to be aware of which channels/bands from which `ImageEntry` in it's set  might link to the RGB bands.

This unfortunately cannot be done automatically for rasters. There is a way to automatically do this on a per-image basis though which I will implement.

**Comment by banesullivan** [comment]
This is ready to go at this point.

-----

there is definitely an issue with the geolocation of the thumbnail, so I will open a new issue about that

**Comment by banesullivan** [comment]
@jtomeck, this PR might have some conflicts with your work

**Comment by banesullivan** [comment]
I can separate the UI stuff from this PR and we can handle that after/during your changes, @jtomeck

**Comment by jtomeck** [comment]
> I can separate the UI stuff from this PR and we can handle that after/during your changes, @jtomeck

Yeah any way you want to do it. If there's conflicts I can always let you know and we can troubleshoot them together if it's something I can't figure out.

**Pull request merged**